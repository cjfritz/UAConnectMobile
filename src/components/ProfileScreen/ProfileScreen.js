import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Card, Icon } from 'react-native-elements';
import {
  View,
  Text,
  Image,
  ImageBackground,
} from 'react-native';
import {
  Item,
  Input,
  Label,
  Button,
  Content,
  Col,
  Container,
} from 'native-base';
import { NavigationEvents } from 'react-navigation';
import { profileUpdate, profileSave, profileFetch } from '../../actions/ProfileActions';
import styles from './ProfileScreen.style';
import backgroundImage from '../../assets/images/ua_logo.png';
import defaultProfile from '../../assets/images/default_profile.jpg';
import { LoadingModal } from '../common/LoadingModal/LoadingModal';

export class ProfileScreen extends Component {
  componentWillMount() {
    const { fetchProfile } = this.props;
    fetchProfile();
  }

  renderHeader = () => {
    const {
      name,
      address,
      phone,
      standing,
      expectedGraduation,
      saveProfile,
    } = this.props;
    return (
      <View style={ styles.headerContainer }>
        <ImageBackground
          style={ styles.headerBackgroundImage }
          blurRadius={ 10 }
          source={ backgroundImage }
        >
          <View style={ styles.headerColumn }>
            <Image
              style={ styles.userImage }
              source={ defaultProfile }
            />
            <Text style={ styles.userNameText }>{ name }</Text>
            <View style={ styles.userAddressRow }>
              <Col>
                <View>
                  <Icon
                    name='place'
                    underlayColor='transparent'
                    iconStyle={ styles.placeIcon }
                  />
                </View>
                <View style={ styles.userCityRow }>
                  <Text style={ styles.userCityText }>
                    { address }
                  </Text>
                </View>
                <Button
                  style={ styles.saveButton }
                  onPress={ () => saveProfile({
                    name,
                    address,
                    phone,
                    standing,
                    expectedGraduation,
                  }) }
                >
                  <Text>
                    Save Changes
                  </Text>
                </Button>
              </Col>
            </View>
          </View>
        </ImageBackground>
      </View>
    );
  };

  renderItem = (prop, propName, title) => {
    const { updateProfile } = this.props;
    return (
      <Item floatingLabel>
        <Label>{ title }</Label>
        <Input
          keyboardType={ propName === 'phone' ? 'phone-pad' : 'default' }
          onChangeText={ text => updateProfile({ prop: propName, value: text }) }
          value={ prop }
        />
      </Item>
    );
  };

  render() {
    const {
      name,
      address,
      phone,
      standing,
      expectedGraduation,
      loading,
      fetchProfile,
    } = this.props;
    return (
      <Container>
        <NavigationEvents
          onWillFocus={ () => {
            fetchProfile();
          } }
        />
        <Content style={ styles.scroll }>
          <LoadingModal visible={ loading } loadingLabel='Loading...' />
          <View style={ styles.container }>
            <Card containerStyle={ styles.cardContainer }>
              { this.renderHeader() }
            </Card>
            <Card>
              { this.renderItem(name, 'name', 'Name') }
            </Card>
            <Card>
              { this.renderItem(address, 'address', 'Address') }
            </Card>
            <Card>
              { this.renderItem(phone, 'phone', 'Phone') }
            </Card>
            <Card>
              { this.renderItem(standing, 'standing', 'Standing') }
            </Card>
            <Card style={ { marginBottom: '5%' } }>
              { this.renderItem(expectedGraduation, 'expectedGraduation', 'Expected Graduation') }
            </Card>

          </View>
        </Content>
      </Container>
    );
  }
}

const mapStateToProps = state => {
  const {
    name,
    address,
    phone,
    standing,
    expectedGraduation,
    loading,
  } = state.profile;
  return {
    name,
    address,
    phone,
    standing,
    expectedGraduation,
    loading,
  };
};

const mapDispatchToProps = dispatch => (
  ({
    updateProfile: ({ prop, value }) => dispatch(profileUpdate({ prop, value })),
    saveProfile: ({
      name,
      address,
      phone,
      standing,
      expectedGraduation,
    }) => dispatch(profileSave({
      name,
      address,
      phone,
      standing,
      expectedGraduation,
    })),
    fetchProfile: () => dispatch(profileFetch()),
  })
);

export default connect(mapStateToProps, mapDispatchToProps)(ProfileScreen);
