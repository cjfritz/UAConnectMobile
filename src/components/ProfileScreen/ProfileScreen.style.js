import { StyleSheet } from 'react-native'; 

export default StyleSheet.create({
  cardContainer: {
    backgroundColor: 'darkred',
    borderWidth: 0,
    flex: 1,
    margin: 0,
    padding: 0,
  },
  container: {
    flex: 1,
  },
  emailContainer: {
    backgroundColor: '#FFF',
    flex: 1,
    paddingTop: 30,
  },
  headerBackgroundImage: {
    paddingBottom: 20,
    paddingTop: 35,
  },
  headerContainer: {},
  headerColumn: {
    alignItems: 'center',
    backgroundColor: 'transparent',
  },
  placeIcon: {
    color: 'white',
    fontSize: 26,
  },
  scroll: {
    backgroundColor: '#FFF',
  },
  telContainer: {
    backgroundColor: '#FFF',
    flex: 1,
    paddingTop: 30,
  },
  userAddressRow: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  userCityRow: {
    backgroundColor: 'transparent',
  },
  userCityText: {
    color: '#A5A5A5',
    fontSize: 15,
    fontWeight: '600',
    textAlign: 'center',
  },
  userImage: {
    borderColor: 'black',
    borderRadius: 85,
    borderWidth: 3,
    height: 170,
    marginBottom: 15,
    width: 170,
  },
  userNameText: {
    color: '#FFF',
    fontSize: 22,
    fontWeight: 'bold',
    paddingBottom: 8,
    textAlign: 'center',
  },
});
