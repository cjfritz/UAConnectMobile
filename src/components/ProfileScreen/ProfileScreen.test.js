import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ProfileScreen } from './ProfileScreen';

const props = {
  loading: false,
  name: '',
  address: '',
  phone: '',
  standing: '',
  expectedGradution: '',
  updateProfile: jest.fn(),
  saveProfile: jest.fn(),
  fetchProfile: jest.fn(),
};

describe('<ProfileScreen />', () => {
  it('should render with empty fields', () => {
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<ProfileScreen { ...props } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should render with complete fields', () => {
    const newProps = {
      name: 'someName',
      address: 'someAddress',
      phone: 'somePhone',
      standing: 'someStanding',
      expectedGraduation: 'someGraduation',
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<ProfileScreen { ...props } />);
    shallowRenderer.render(<ProfileScreen { ...props } { ...newProps } />);
    expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
    expect(props.fetchProfile).toHaveBeenCalled();
  });
});
