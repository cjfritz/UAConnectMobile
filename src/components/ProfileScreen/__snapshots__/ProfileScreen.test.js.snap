// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileScreen /> should render with complete fields 1`] = `
<Styled(Container)>
  <withNavigation(NavigationEvents)
    onWillFocus={[Function]}
  />
  <Styled(Content)
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  >
    <LoadingModal
      loadingLabel="Loading..."
      visible={false}
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Card
        containerStyle={
          Object {
            "backgroundColor": "darkred",
            "borderWidth": 0,
            "flex": 1,
            "margin": 0,
            "padding": 0,
          }
        }
      >
        <Component>
          <ImageBackground
            blurRadius={10}
            source={
              Object {
                "testUri": "../../../src/assets/images/ua_logo.png",
              }
            }
            style={
              Object {
                "paddingBottom": 20,
                "paddingTop": 35,
              }
            }
          >
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                }
              }
            >
              <Component
                source={
                  Object {
                    "testUri": "../../../src/assets/images/default_profile.jpg",
                  }
                }
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "black",
                    "borderRadius": 85,
                    "borderWidth": 3,
                    "height": 170,
                    "marginBottom": 15,
                    "width": 170,
                  }
                }
              />
              <Component
                style={
                  Object {
                    "color": "#FFF",
                    "fontSize": 22,
                    "fontWeight": "bold",
                    "paddingBottom": 8,
                    "textAlign": "center",
                  }
                }
              >
                someName
              </Component>
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <ColumnNB>
                  <Component>
                    <Icon
                      color="black"
                      iconStyle={
                        Object {
                          "color": "white",
                          "fontSize": 26,
                        }
                      }
                      name="place"
                      raised={false}
                      reverse={false}
                      reverseColor="white"
                      size={24}
                      underlayColor="transparent"
                    />
                  </Component>
                  <Component
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "paddingBottom": "2%",
                      }
                    }
                  >
                    <Component
                      style={
                        Object {
                          "color": "white",
                          "fontSize": 15,
                          "fontWeight": "600",
                          "textAlign": "center",
                        }
                      }
                    >
                      someAddress
                    </Component>
                  </Component>
                  <Styled(Button)
                    onPress={[Function]}
                    style={
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "white",
                      }
                    }
                  >
                    <Component>
                      Save Changes
                    </Component>
                  </Styled(Button)>
                </ColumnNB>
              </Component>
            </Component>
          </ImageBackground>
        </Component>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Name
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
            value="someName"
          />
        </Styled(Item)>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Address
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
            value="someAddress"
          />
        </Styled(Item)>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Phone
          </Styled(Label)>
          <Styled(Input)
            keyboardType="phone-pad"
            onChangeText={[Function]}
            value="somePhone"
          />
        </Styled(Item)>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Standing
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
            value="someStanding"
          />
        </Styled(Item)>
      </Card>
      <Card
        style={
          Object {
            "marginBottom": "5%",
          }
        }
      >
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Expected Graduation
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
            value="someGraduation"
          />
        </Styled(Item)>
      </Card>
    </Component>
  </Styled(Content)>
</Styled(Container)>
`;

exports[`<ProfileScreen /> should render with empty fields 1`] = `
<Styled(Container)>
  <withNavigation(NavigationEvents)
    onWillFocus={[Function]}
  />
  <Styled(Content)
    style={
      Object {
        "backgroundColor": "#FFF",
      }
    }
  >
    <LoadingModal
      loadingLabel="Loading..."
      visible={false}
    />
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Card
        containerStyle={
          Object {
            "backgroundColor": "darkred",
            "borderWidth": 0,
            "flex": 1,
            "margin": 0,
            "padding": 0,
          }
        }
      >
        <Component>
          <ImageBackground
            blurRadius={10}
            source={
              Object {
                "testUri": "../../../src/assets/images/ua_logo.png",
              }
            }
            style={
              Object {
                "paddingBottom": 20,
                "paddingTop": 35,
              }
            }
          >
            <Component
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                }
              }
            >
              <Component
                source={
                  Object {
                    "testUri": "../../../src/assets/images/default_profile.jpg",
                  }
                }
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "black",
                    "borderRadius": 85,
                    "borderWidth": 3,
                    "height": 170,
                    "marginBottom": 15,
                    "width": 170,
                  }
                }
              />
              <Component
                style={
                  Object {
                    "color": "#FFF",
                    "fontSize": 22,
                    "fontWeight": "bold",
                    "paddingBottom": 8,
                    "textAlign": "center",
                  }
                }
              >
                
              </Component>
              <Component
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <ColumnNB>
                  <Component>
                    <Icon
                      color="black"
                      iconStyle={
                        Object {
                          "color": "white",
                          "fontSize": 26,
                        }
                      }
                      name="place"
                      raised={false}
                      reverse={false}
                      reverseColor="white"
                      size={24}
                      underlayColor="transparent"
                    />
                  </Component>
                  <Component
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "paddingBottom": "2%",
                      }
                    }
                  >
                    <Component
                      style={
                        Object {
                          "color": "white",
                          "fontSize": 15,
                          "fontWeight": "600",
                          "textAlign": "center",
                        }
                      }
                    >
                      
                    </Component>
                  </Component>
                  <Styled(Button)
                    onPress={[Function]}
                    style={
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "white",
                      }
                    }
                  >
                    <Component>
                      Save Changes
                    </Component>
                  </Styled(Button)>
                </ColumnNB>
              </Component>
            </Component>
          </ImageBackground>
        </Component>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Name
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
            value=""
          />
        </Styled(Item)>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Address
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
            value=""
          />
        </Styled(Item)>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Phone
          </Styled(Label)>
          <Styled(Input)
            keyboardType="phone-pad"
            onChangeText={[Function]}
            value=""
          />
        </Styled(Item)>
      </Card>
      <Card>
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Standing
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
            value=""
          />
        </Styled(Item)>
      </Card>
      <Card
        style={
          Object {
            "marginBottom": "5%",
          }
        }
      >
        <Styled(Item)
          floatingLabel={true}
        >
          <Styled(Label)>
            Expected Graduation
          </Styled(Label)>
          <Styled(Input)
            keyboardType="default"
            onChangeText={[Function]}
          />
        </Styled(Item)>
      </Card>
    </Component>
  </Styled(Content)>
</Styled(Container)>
`;
